tcping.go

```diff
@@ -410,11 +421,28 @@ func resolveHostname(tcpStats *stats) ipAddress {
 
 /* Retry resolve hostname after certain number of failures */
 func retryResolve(tcpStats *stats) {
-	if tcpStats.ongoingUnsuccessfulProbes >= tcpStats.retryHostnameResolveAfter {
-		tcpStats.printer.printRetryingToResolve(tcpStats.hostname)
-		tcpStats.ip = resolveHostname(tcpStats)
-		tcpStats.ongoingUnsuccessfulProbes = 0
-		tcpStats.retriedHostnameResolves += 1
+	// If we have a limit set (-r) and not reached it yet, just continue pinging.
+	if tcpStats.retryHostnameResolveAfter != 0 &&
+		tcpStats.ongoingUnsuccessfulProbes < tcpStats.retryHostnameResolveAfter {
+		return
+	}
+
```

Comment:
I prefer the previous condition check more.

This way, we have two returns in the body of the function whereas the previous one was just a simple check.

I know we are usually taught to avoid indentation by introducing early returns but I am a big proponent of a code that reads like English and trust the compiler to iron out the details.

The rest of the code is very good. We can merge it in after this one is addressed.

Thanks!